jubilant/_all_any.py

```diff
@@ -50,6 +50,38 @@ def all_error(status: Status, *apps: str) -> bool:
     return _all_statuses_are('error', status, apps)
 
 
+def all_idle_agents(status: Status, *apps: str) -> bool:
+    """Report whether all unit agents in *status* (filtered to *apps* if provided) are "idle".
+
+    Unlike the other ``all_*`` and ``any_*`` helpers, this method looks at the status of each
+    Juju unit agent, not the workload's application or unit status.
+
+    Examples::
+
+        # Use the callable directly to wait for unit agents from all apps to be idle.
+        juju.wait(jubilant.all_idle_agents)
+
+        # Use a lambda to wait for unit agents only from specified apps (blog, mysql).
+        juju.wait(lambda status: jubilant.all_idle_agents(status, 'blog', 'mysql'))
+
+    Args:
+        status: The status object being tested.
+        apps: An optional list of application names. If provided, only the unit agents of these
+            applications are tested, and each must be present in ``status.apps``.
+    """
+    if not apps:
+        apps = tuple(status.apps.keys())
```

Comment:
Is there a reason to not just have `apps = status.apps` here? Or `apps = tuple(status.apps)` if you want a copy for some reason?

Reply:
There's no reason to use `.keys()` -- thanks, I've removed that. The `tuple(...)` is to satisfy the type checker. But I've actually refactored this to use a helper function like the other `all_*` methods, and it's better and more consistent that way.